indexing-service/src/main/java/io/druid/indexing/common/task/AbstractTask.java

```diff
@@ -179,4 +183,24 @@ public int hashCode()
   {
     return toolbox.getTaskActionClient().submit(new LockListAction());
   }
+
+  @Override
+  @JsonProperty
+  public Map<String, Object> getContext()
+  {
+    return context;
+  }
+
+  @Override
+  public <ContextType> ContextType getContextValue(String key)
+  {
+    return context == null ? null : (ContextType) context.get(key);
```

Comment:
I used the same semantics as used by druid queries.
I like this way so that user of this method need not worry about casting the returned value. 
I can change it to Object as return type if you feel strongly about it ?